src/sphinx_autodoc_typehints/__init__.py

```diff
@@ -89,7 +89,19 @@ def get_annotation_args(annotation: Any, module: str, class_name: str) -> tuple[
     return getattr(annotation, "__args__", ())
 
 
-def format_annotation(annotation: Any, fully_qualified: bool = False, simplify_optional_unions: bool = True) -> str:
+def format_annotation(
+    annotation: Any,
+    fully_qualified: bool = False,
+    simplify_optional_unions: bool = True,
+    typehints_formatter: Callable[..., str] | None = None,
+) -> str:
+    if typehints_formatter is not None:
+        formatted = typehints_formatter(
+            annotation, fully_qualified=fully_qualified, simplify_optional_unions=simplify_optional_unions
```

Comment:
Finish this with a comma to force split into new lines (to be consistent with other instances)